### PR TITLE
Stripe tx show page fixes

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

**Comment by maxwofford** [comment]
This PR makes a whole heap of UI changes, so I'll list them here:

1. Now declined TXs will show up in your org's transaction list for 24 hours after the charge is made:
    Screen Shot 2022-04-11 at 15 40 45
1. Stripe card charges have more details on their HCB show page, such as card number, & the name of the merchant
    Screen Shot 2022-04-11 at 15 44 04
    Screen Shot 2022-04-11 at 15 49 12
1. The show page also includes new modals for merchant or charge verification info:
    Screen Shot 2022-04-11 at 15 46 09
    Screen Shot 2022-04-11 at 15 46 15

**Comment by maxwofford** [comment]
Hey @garyhtou @exu3, assigning you two for review b/c this touches a lot of stuff & wanted to get your approval before merging.

Pages changed:
- Event show page (specifically, the list of TXs that show up now include declined stripe charges in the past 24 hours)
	- this includes transparency mode
- HCB show pages for stripe card charges
	- these will look different depending on a charge being declined, pending, or settled, so check out a couple different ones.

**Comment by maxwofford** [comment]
I'd like to add a changelog post about this & ship sometime this week. Are there any other UI quick wins we can add in to this that don't distract from our other work? I'm thinking the messaging could be "it's now easier to get context on a given TX, and find out why something was approved/declined".

**Comment by maxwofford** [comment]
Didn't assign for review, but cc @cjdenio too!

**Comment by garyhtou** [comment]
head up, i probably don't be able to take a look at this until next week. sorry!

File: app/models/canonical_pending_transaction.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ class CanonicalPendingTransaction < ApplicationRecord
       .includes(:canonical_pending_declined_mappings)
       .where(canonical_pending_declined_mappings: { canonical_pending_transaction_id: nil })
   }
+  scope :unsettled_with_declined, -> { includes(:canonical_pending_settled_mappings).where(created_at: (Time.now - 24.hours)..Time.now, canonical_pending_settled_mappings: { canonical_pending_transaction_id: nil }) }
```
Comment (garyhtou) [inline app/models/canonical_pending_transaction.rb]: kinda picky:

```suggestion
  scope :unsettled_with_declined, -> { includes(:canonical_pending_settled_mappings).where(created_at: 24.hours.ago..Time.now, canonical_pending_settled_mappings: { canonical_pending_transaction_id: nil }) }
```

File: app/models/canonical_pending_transaction.rb [hunk 2]
```diff
@@ -40,6 +40,7 @@ class CanonicalPendingTransaction < ApplicationRecord
       .includes(:canonical_pending_declined_mappings)
       .where(canonical_pending_declined_mappings: { canonical_pending_transaction_id: nil })
   }
+  scope :unsettled_with_declined, -> { includes(:canonical_pending_settled_mappings).where(created_at: 24.hours.ago..Time.now, canonical_pending_settled_mappings: { canonical_pending_transaction_id: nil }) }
```
Comment (garyhtou) [inline app/models/canonical_pending_transaction.rb]: @maxwofford is there a reason we're limiting unsettled transactions to 24 hours ago? it seems to be hiding some transactions and creating an inconsistent presentation of data to users

example:
https://bank.hackclub.com/sc-office-365-productivity-group

$350 incoming, but no pending transactions on dashboard
image

console reveals a pending tx:
image

Comment (maxwofford) [reply app/models/canonical_pending_transaction.rb]: I think I wrote that line wrong– I'm trying to match unsettled, but also include declined under a day old

Comment (garyhtou) [reply app/models/canonical_pending_transaction.rb]: Ah okay, try taking a look at `CanonicalPendingDeclinedMapping`

Comment (maxwofford) [reply app/models/canonical_pending_transaction.rb]: The highlighted event doesn't touch the `unsettled_with_declined` scope as far as I can tell. I think that's another bug.

Comment (maxwofford) [reply app/models/canonical_pending_transaction.rb]: Errr... it does, but it's not touching the pending incoming balance bug– that's caused by invalid data. Here it is with the 24 hour limit turned off:

Screen Shot 2022-04-19 at 12 32 52
 
That pending invoice is from almost 16 months ago, which means it probably shouldn't be showing up.

**Pull request merged**